app/controllers/matter_controller.rb

```diff
@@ -2,6 +2,8 @@
 
 class MatterController < ApplicationController
 
+  before_action :require_manage_token
+
```

Comment:
Instead of defining `manage_token` as a method and dealing with when it's blank, we can use the before_action pattern, which lets us redirect to the homepage and skip this controller's actions altogether.